sfge/src/main/java/com/salesforce/graph/ops/TraversalUtil.java

```diff
@@ -68,5 +70,64 @@ public static GraphTraversal<Vertex, Vertex> ruleTargetTraversal(
                                         .toArray(GraphTraversal[]::new));
     }
 
+    public static GraphTraversal<Vertex, Vertex> traverseImplementationsOf(
+            GraphTraversalSource g, List<String> targetFiles, String interfaceName) {
+        // For our traversal, we want to start with every class that implements either the target
+        // interface or one of its
+        // subtypes. Do that with a union of these two subtraversals.
+        // Also, start with the hasLabel() call, because doing an initial filter along an indexed
+        // field saves us a ton of time.
+        GraphTraversal<Vertex, Vertex> traversal =
+                g.V().hasLabel(NodeType.USER_CLASS, NodeType.USER_INTERFACE)
```

Comment:
This `hasLabel` call isn't technically necessary, but adding it improved the performance of the query substantially.